library/Control/Monad/AWS/ViaMock.hs

```diff
@@ -76,3 +80,14 @@ newtype MockAWS m a = MockAWS
 instance (MonadIO m, MonadReader env m, HasMatchers env) => MonadAWS (MockAWS m) where
   sendEither = matchSend
   awaitEither = matchAwait
+  withAuth = ($ fakeAuthEnv)
+  modified _ = id
```

Comment:
Would it be possible to go all the way one direction or another here:

- Operations involving `AuthEnv` are not supported by the mock (`withAuth = undefined`)
- `AuthEnv` is supported by the mock (`newtype MockAWS m a = MockAWS (AuthEnv -> m a)`) and `modified` works as normal based on `local`

Reply:
Nevermind, I was confusing `Env` and `AuthEnv` so I thought `withAuth` and `modified` were more related than they are.

Reply:
The naming there is not the best for sure.